components/collapsible-panel/collapsible-panel.js

```diff
@@ -0,0 +1,344 @@
+import '../colors/colors.js';
+import '../icons/icon.js';
+import { css, html, LitElement } from 'lit';
+import { heading1Styles, heading2Styles, heading3Styles, heading4Styles } from '../typography/styles.js';
+import { hide, show } from '../../directives/animate/animate.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+// TODO: should we export/share this? I stole this from calendar
+const keyCodes = {
+	ENTER: 13,
+	SPACE: 32,
+};
+
+/**
+ * A container with a title that can be expanded/collapsed to show/hide content.
+ * @slot header - Slot for header content, such as course image (no actionable elements)
+ * @slot summary - Slot for the summary of the expanded content. Only accepts `d2l-collapsible-panel-summary-item`
+ * @slot default - Slot for the expanded content
+ * @slot actions - Slot for buttons and dropdown openers to be placed in top right corner of header
+ * @fires d2l-collapsible-panel-expand - Dispatched when the panel is expanded
+ * @fires d2l-collapsible-panel-collapse - Dispatched when the panel is collapsed
+ */
+class CollapsiblePanel extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * REQUIRED: The title of the panel
+			 * @type {string}
+			 */
+			title: { type: String },
+			/**
+			 * The semantic heading level (h1-h5)
+			 * @type {'1'|'2'|'3'|'4'|'5'}
+			 * @default "3"
+			 */
+			headingLevel: { attribute: 'heading-level', type: String, reflect: true },
+			/**
+			 * The heading style to use
+			 * @type {'1'|'2'|'3'|'4'|'5'}
+			 * @default "3"
+			 */
+			headingStyle: { attribute: 'heading-style', type: String, reflect: true },
+			/**
+			 * Whether or not the panel is expanded
+			 * @type {boolean}
+			 */
+			expanded: { type: Boolean, reflect: true },
+			/**
+			 * REQUIRED: Label describing the contents of the header.
+			 * Used for screen readers.
+			 * @type {string}
+			 */
+			expandCollapseLabel: { attribute: 'expand-collapse-label', type: String, reflect: true },
+			/**
+			 * Type of collapsible panel
+			 * @type {'default'|'subtle'|'inline'}
+			 * @default "default"
+			 */
+			type: { type: String, reflect: true },
+			/**
+			 * Whether or not the content should extend the full width.
+			 * This is only valid when using the "inline" panel style.
+			 * @type {boolean}
+			 */
+			fullWidth: { attribute: 'full-width', type: Boolean, reflect: true },
+			_hasSummary: { type: Boolean, reflect: true },
```

Comment:
This isn't a required change per se... but reflecting these private properties is kinda leaky. In effect, it is leaking them as attributes on the host, which is public. We _do_ have several components that do this, especially when the CSS is targeting the host (ex. `:host([some-attr]) { ... }`) but when the target is an internal element then we usually try to avoid leaking by rendering an internal class, which means the CSS changes from:

```css
:host([some-attr]) .some-elem {
    ...
}
```
to the following:
```css
.some-elem.some-class {
    ...
}
```

With such a change, the property definition can become `_hasSummary: { state: true },` since reflection is no longer necessary.